ci-operator/step-registry/mirror-images/by-oc-mirror/mirror-images-by-oc-mirror-commands.sh

```diff
@@ -98,52 +91,52 @@ extract_oc_mirror "${new_pull_secret}" "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRI
-    echo "${readable_version} is found in ${channel_name} on ${ocp_image_arch} arch"
-else
-    echo "${readable_version} is NOT found in ${channel_name} on ${ocp_image_arch} arch, exit..."
-    exit 1
-fi
 
 # set the imagesetconfigure
 image_set_config="image_set_config.yaml"
 cat <<END | tee "${image_set_config}"
 kind: ImageSetConfiguration
-apiVersion: mirror.openshift.io/v1alpha2
+apiVersion: mirror.openshift.io/v2alpha1
 mirror:
   platform:
-    architectures:
-      - "${ocp_image_arch}"
-    channels:
-      - name: ${channel_name}
-        minVersion: ${readable_version}
-        maxVersion: ${readable_version}
+    release: ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}
 END
 
 # https://github.com/openshift/oc-mirror/blob/main/docs/usage.md#authentication
 # oc-mirror only respect ~/.docker/config.json -> ${XDG_RUNTIME_DIR}/containers/auth.json
 mkdir -p "${XDG_RUNTIME_DIR}/containers/"
 cp -rf "${new_pull_secret}" "${XDG_RUNTIME_DIR}/containers/auth.json"
 
+unset REGISTRY_AUTH_PREFERENCE
+# execute the oc-mirror command with dry-run
+dryruncmd="'${oc_mirror_bin}' -c ${image_set_config} docker://${target_release_image_repo} --dest-tls-verify=false --v2 --workspace file://${mirror_output} --dry-run"
+eval "$dryruncmd" 
+
 # execute the oc-mirror command
-cmd="'${oc_mirror_bin}' -c ${image_set_config} docker://${target_release_image_repo} --dest-skip-tls"
+cmd="'${oc_mirror_bin}' -c ${image_set_config} docker://${target_release_image_repo} --dest-tls-verify=false --v2 --workspace file://${mirror_output}"
 echo "Command: $cmd"
 eval "$cmd" | tee "${mirror_output}"
 
 # Get mirror setting for install-config.yaml
-result_folder=$(cat "${mirror_output}" | tail -n 1 | awk  '{print $NF}')
-icsp_file="${result_folder}/imageContentSourcePolicy.yaml"
-mapping_file="${result_folder}/mapping.txt"
-if [ ! -f "${icsp_file}" ]; then
-    echo "${icsp_file} not found, exit..."
+result_folder="${mirror_output}/working-dir"
+idms_file="${result_folder}/cluster-resources/idms-oc-mirror.yaml"
+itms_file="${result_folder}/cluster-resources/itms-oc-mirror.yaml"
+mapping_file="${result_folder}/dry-run/mapping.txt"
+if [ ! -f "${idms_file}" ]; then
+    echo "${idms_file} not found, exit..."
     exit 1
 fi
 key_name="imageContentSources"
 if [[ "${ENABLE_IDMS}" == "yes" ]]; then
     key_name="imageDigestSources"
 fi
-yq-v4 --prettyPrint eval-all "{\"$key_name\": .spec.repositoryDigestMirrors}" "${icsp_file}" > "${install_config_mirror_patch}" || exit 1
+
+yq --prettyPrint eval-all "{\"$key_name\": .spec.imageDigestMirrors}" "${idms_file}" > "${install_config_mirror_patch}" || exit 1
+if [ -f "${itms_file}" ]; then
+    echo "${itms_file} not found, exit..."
+    yq --prettyPrint eval-all "{\"\": .spec.imageTagMirrors}" "${itms_file}" >> "${install_config_mirror_patch}" || exit 1
+fi
+sed -i "5d" "${install_config_mirror_patch}"
```

Comment:
What purpose for the line ?

Reply:
Does the superfluous line affect the installation ?

Reply:
cat /tmp/install_config_patch 
imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev

Reply:
cat /tmp/install_config_patch 
`imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev`

Reply:
cat /tmp/install_config_patch 
"imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev"

Reply:
cat /tmp/install_config_patch 
```
imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
imageContentSources:
  - mirrors:
      - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
```

Reply:
[fedora@preserve-fedora-yinzhou yinzhou]$ cat nn/working-dir/cluster-resources/idms-oc-mirror.yaml 
---
```
apiVersion: config.openshift.io/v1
kind: ImageDigestMirrorSet
metadata:
  name: idms-release-0
spec:
  imageDigestMirrors:
  - mirrors:
    - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
status: {}
```
[fedora@preserve-fedora-yinzhou yinzhou]$ cat nn/working-dir/cluster-resources/itms-oc-mirror.yaml 
---
```
apiVersion: config.openshift.io/v1
kind: ImageTagMirrorSet
metadata:
  name: itms-release-0
spec:
  imageTagMirrors:
  - mirrors:
    - localhost:5000/openshift-release-dev
    source: quay.io/openshift-release-dev
status: {}
```

Reply:
the idms and itms are same here .

Reply:
Hmm, if that, why not directly append `spec.imageTagMirrors` from itms to "${install_config_mirror_patch}"